app/controllers/revisions_controller.rb

```diff
@@ -57,12 +57,43 @@ def latest_diff
     end
   end
 
+  # PATCH /notebooks/:notebook_id/revisions/:commit_id/edit_friendly_label
+  def edit_friendly_label
+    errors = ""
+    notebook = Notebook.find(@revision.notebook_id)
+    friendly_label = params[:friendly_label].strip
+    if verify_revision_label(friendly_label, notebook, params[:old_label])
```

Comment:
You are calling the same function twice with the same parameters.  Save the result the first time you call it and then append it to the errors. Don't call it twice.

Reply:
Thank you. Good catch. My mistake. Fixed.